src/VecSim/algorithms/svs/svs.h

```diff
@@ -480,6 +515,18 @@ class SVSIndex : public VecSimIndexAbstract<svs_details::vecsim_dt<DataType>, fl
         return res;
     }
 
+    void runGC() override {
+        if (impl_) {
+            // There is documentation for consolidate():
+            // https://intel.github.io/ScalableVectorSearch/python/dynamic.html#svs.DynamicVamana.consolidate
+            impl_->consolidate();
```

Comment:
The documentation says "Should be called after a sufficient number of deletions to avoid the memory consumption of the index monotonically increasing."
Is this enforced? if so, where?

Reply:
but here it is invoked with no check on "sufficient number of deletions" ?

Reply:
Yes, as far as it is 'Garbage Collection' routine, `consolidate()` is called unconditionally.

Reply:
It might be my confusion, but `runGC()` is declared in the `VecSimIndexInterface` so, my assumption was that any index can implement this method according to it's internal design.
In addition, this allows to follow incapsulation rules for `SVSIndex` and prevents:
1. Exposing `impl_` field with `public:` access.
2. Avoid direct dependency on `SVSIndex` template parameters in `SVSTieredIndex`.
3. Simplify `SVSTieredIndex` definition: no extra template parameters needed to use `SVSIndex`

It was also possible to declare a special utility method in `SVSIndexBase`, but I assumed that reusing existing one is allowed.

Reply:
The original intention was that only tiered indexes would implement `runGC()`, and the reason that is part of `VecSimIndexInterface` is to expose it in the C API. In practice, `VecSimIndexAbstract` overrides `runGC()` with an empty implementation:
```
void runGC() override {}              // Do nothing, relevant for tiered index only.
```
However, since in svs there is no notion of "delete in place" and running this GC is required even in non-tiered index. I also understand your points regarding encapsulation, so let's leave it as it is.